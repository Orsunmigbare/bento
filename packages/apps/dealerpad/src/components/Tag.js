import React from "react"
import styled from "styled-components"
import { spacing, color, targetSize, font } from "@staccx/theme"

const Tag = ({ children, className, ...rest }) => (
  <TagWrapper className={className} {...rest}>
    {children}
  </TagWrapper>
)

const TagWrapper = styled.span`
  display: inline-block;
  font-size: ${font.tiny};
  text-align: center;
  padding: calc(${spacing.tiny} / 2) calc(${spacing.medium} / 1.5);
  border-radius: ${targetSize.normal};
  background-color: ${color.white};
  color: ${color.primary};
`

export default Tag
