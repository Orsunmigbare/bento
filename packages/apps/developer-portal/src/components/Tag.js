import React from "react"
import PropTypes from "prop-types"
import { ExpandListItem } from "@staccx/base"
import SchemaConsumer from "./SchemaConsumer"
import Paths from "./Paths"

class Tag extends React.Component {
  render() {
    const { tag } = this.props
    if (!tag) {
      return null
    }

    const name = typeof tag === "string" ? tag : tag.name

    return (
      <SchemaConsumer>
        {({ sorted }) => {
          return (
            <ExpandListItem title={name}>
              <Paths paths={sorted[name]} />
            </ExpandListItem>
          )
        }}
      </SchemaConsumer>
    )
  }
}

export default Tag
