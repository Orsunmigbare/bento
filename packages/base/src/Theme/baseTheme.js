import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
