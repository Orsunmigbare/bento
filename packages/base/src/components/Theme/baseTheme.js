export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
