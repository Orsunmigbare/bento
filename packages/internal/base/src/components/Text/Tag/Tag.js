import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  spacing,
  color,
  targetSize,
  font,
  themify,
  themeProps
} from "@staccx/theme"
import themePropTypes from "../../constants/themePropTypes"

const Tag = ({ children, className, variant, ...rest }) => (
  <TagWrapper className={className} variant={variant} {...rest}>
    {children}
  </TagWrapper>
)

Tag.themeProps = {
  tag: {
    name: "tag",
    description: "A span with padding around it",
    type: themePropTypes.style
  }
}

const TagWrapper = styled.span`
  display: inline-block;
  font-size: ${font.tiny};
  text-align: center;
  padding: calc(${spacing.tiny} / 2) calc(${spacing.medium} / 1.5);
  border-radius: ${targetSize.normal};
  background-color: ${color.subtleHover};
  color: ${color.primary};
  min-height: ${spacing.medium};
  ${themify(Tag.themeProps.tag)};
`

Tag.defaultProps = {
  className: null
}

Tag.propTypes = {
  children: themeProps.children.isRequired,
  className: PropTypes.string,
  variant: PropTypes.string
}

export default Tag
