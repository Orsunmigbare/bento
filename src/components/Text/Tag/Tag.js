import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  themePropTypes,
  commonPropTypes
} from "../../../constants/themeContants"
import {
  applyVariants,
  font,
  color,
  spacing,
  targetSize
} from "../../../theming"

const Tag = ({ children, className, variant, ...rest }) => (
  <TagWrapper className={className} variant={variant} {...rest}>
    {children}
  </TagWrapper>
)

Tag.themeProps = {
  tag: {
    name: "tag",
    description: "A span with padding around it",
    type: themePropTypes.style
  }
}

const TagWrapper = styled.span`
  display: inline-block;
  font-size: ${font.tiny};
  text-align: center;
  padding: calc(${spacing.tiny} / 2) calc(${spacing.medium} / 1.5);
  border-radius: ${targetSize.normal};
  background-color: ${color.subtleHover};
  color: ${color.primary};
  min-height: ${spacing.medium};
  ${applyVariants(Tag.themeProps.tag)};
`

Tag.defaultProps = {
  className: null
}

Tag.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string,
  variant: PropTypes.string
}

export default Tag
