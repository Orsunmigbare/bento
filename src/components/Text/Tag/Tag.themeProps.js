import { themePropTypes } from "../../../theming"

const themeProps = {
  tag: {
    name: "tag",
    description: "A span with padding around it",
    type: themePropTypes.style
  }
}

export default themeProps
