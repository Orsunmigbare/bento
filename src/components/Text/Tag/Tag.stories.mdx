import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Tag from "./Tag"

<Meta title="Components|Text/Tag" component={Tag}/>

# Tag
Tag are used to add content and additional info to objects.

```javascript
import { Tag } from "@staccx/bento"
```

## Standard Tag
This is the standard Tag
<Preview>
  <Story name="Standard">
   <Tag>This is a tag</Tag>
  </Story>
</Preview>


## Props for Tag
<Props of={Tag} />
