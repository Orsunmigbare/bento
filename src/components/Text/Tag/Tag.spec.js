import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Tag from "./Tag"

describe("Tag", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Tag>This is a tag</Tag>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("no Children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Tag />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("ClassName", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Tag className="className">hei</Tag>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
