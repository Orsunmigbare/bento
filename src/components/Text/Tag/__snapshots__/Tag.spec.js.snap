// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Snapshots ClassName 1`] = `
<span
  className="sc-AykKC jifFQv className"
>
  hei
</span>
`;

exports[`Tag Snapshots No props 1`] = `
<span
  className="sc-AykKC jifFQv"
>
  This is a tag
</span>
`;

exports[`Tag Snapshots no Children 1`] = `
<span
  className="sc-AykKC jifFQv"
/>
`;
